variables.tf

```diff
@@ -107,3 +107,100 @@ variable "pi_tags" {
+      file_name: string, full file name of the image inside COS bucket
+      storage_tier: string, storage tier which the image will be stored in after import. Must be one of the storage tiers supported in the PowerVS workspace region. Available tiers can be found using `ibmcloud pi storage-tiers`. Typical values are: "tier0", "tier1", "tier3", "tier5k"
+      sap_type: string, "Hana", "Netweaver". Set to null if it's not an SAP image.
+  EOF
+  type = object({
+    image_name   = string
+    file_name    = string
+    storage_tier = string
+    sap_type     = string
+  })
+  validation {
+    condition     = var.pi_custom_image2 != null ? var.pi_custom_image2.sap_type == null ? true : contains(["Hana", "Netweaver"], var.pi_custom_image2.sap_type) : true
+    error_message = "Unsupported sap_type in pi_custom_image2. Supported values: null, \"Hana\", \"Netweaver\"."
+  }
+  default = null
+}
+
+variable "pi_custom_image3" {
+  description = <<EOF
+    Optional custom image to import from Cloud Object Storage into PowerVS workspace.
+      image_name: string, must be unique image name how the image will be named inside PowerVS workspace
+      file_name: string, full file name of the image inside COS bucket
+      storage_tier: string, storage tier which the image will be stored in after import. Must be one of the storage tiers supported in the PowerVS workspace region. Available tiers can be found using `ibmcloud pi storage-tiers`. Typical values are: "tier0", "tier1", "tier3", "tier5k"
+      sap_type: string, "Hana", "Netweaver". Set to null if it's not an SAP image.
+  EOF
+  type = object({
+    image_name   = string
+    file_name    = string
+    storage_tier = string
+    sap_type     = string
+  })
+  validation {
+    condition     = var.pi_custom_image3 != null ? var.pi_custom_image3.sap_type == null ? true : contains(["Hana", "Netweaver"], var.pi_custom_image3.sap_type) : true
+    error_message = "Unsupported sap_type in pi_custom_image3. Supported values: null, \"Hana\", \"Netweaver\"."
+  }
+  default = null
+}
+
+variable "pi_custom_image_cos_configuration" {
+  description = <<EOF
+    Cloud Object Storage bucket containing the custom PowerVS images. Images will be imported into the PowerVS Workspace.
+      bucket_name: string, name of the COS bucket
+      bucket_access: string, possible values: "public", "private" (private requires pi_custom_image_cos_service_credentials)
+      bucket_region: string, COS bucket region
+  EOF
+  type = object({
+    bucket_name   = string
+    bucket_access = string
+    bucket_region = string
+  })
+  default = null
+  validation {
+    condition     = var.pi_custom_image_cos_configuration != null ? contains(["public", "private"], var.pi_custom_image_cos_configuration.bucket_access) : true
+    error_message = "Invalid pi_custom_image_cos_configuration.bucket_access. Allowed values: [\"public\", \"private\"]."
+  }
+  validation {
```

Comment:
I don't think this possible. Please check. Cannot access another variable inside validation. Will have to move this validation to local variable inside pi_images.tf file

Reply:
I made a logic error in the validation and fixed it. I tried it and it now validates correctly. If all images are null it doesn't throw an error but if at least one is defined, it throws an error.
<img width="897" alt="Bildschirmfoto 2024-10-21 um 16 02 50" src="https://github.com/user-attachments/assets/62cf5d68-39d3-4605-8bca-407931581692">